vagrant/logs-keeper/logs-keeper.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+import logging
+import os
+import shutil
+import sys
+import time
+
+USAGE="""
+Log Keeper follows all symlinks in given directory and creates hardlinks to prevent data loss because of file rotation
+
+For example for given directory structure:
+
+tree /logs-keeper --inodes
+logs-keeper
+├── [5774569]  containers
+│   └── [5774571]  example_file_in_containers.json -> /logs-keeper/docker/example_file.json
+└── [5774570]  docker
+    └── [5774571]  example_file.json
+
+and <directory_path> set to /logs-keeper/containers, logs-keeper is going to create some additional directories:
+
+tree /logs-keeper --inodes
+logs-keeper
+├── [5774569]  containers
+│   ├── [5774571]  example_file_in_containers.json -> /logs-keeper/docker/example_file.json
+│   └── [5774580]  sumologic
+│       └── [5774581]  5774571
+│           └── [5774571]  example_file.json -> /logs-keeper/docker/sumologic/5774571/example_file_in_containers.json
+└── [5774570]  docker
+    ├── [5774571]  example_file.json
+    └── [5774574]  sumologic
+        └── [5774578]  5774571
+            └── [5774571]  example_file_in_containers.json
+
+It created sumologic directory in both places (where the symlink is and where the link target is).
+In this directory another one is created and named with inode value
+
+Below rotation scenario is presented.
+
+Moment of rotation:
+
+tree /logs-keeper --inodes
+logs-keeper
+├── [5774569]  containers
+│   ├── [5774571]  example_file_in_containers.json -> /logs-keeper/docker/example_file.json
+│   └── [5774580]  sumologic
+│       └── [5774581]  5774571
+│           └── [5774571]  example_file.json -> /logs-keeper/docker/sumologic/5774571/example_file_in_containers.json
+└── [5774570]  docker
+    ├── [5774572]  example_file.json
+    ├── [5774571]  example_file.json.1
+    └── [5774574]  sumologic
+        └── [5774578]  5774571
+            └── [5774571]  example_file_in_containers.json
+
+Handling rotation by logs-keeper:
+
+tree /logs-keeper --inodes
+logs-keeper
+├── [5774569]  containers
+│   ├── [5774571]  example_file_in_containers.json -> /logs-keeper/docker/example_file.json
+│   └── [5774580]  sumologic
+│       |── [5774581]  5774571
+│           └── [5774571]  example_file.json -> /logs-keeper/docker/sumologic/5774571/example_file_in_containers.json
+│       └── [5774576]  5774572
+│           └── [5774572]  example_file.json -> /logs-keeper/docker/sumologic/5774572/example_file_in_containers.json
+└── [5774570]
+    ├── [5774572]  example_file.json
+    ├── [5774571]  example_file.json.1
+    └── [5774574]  sumologic
+        ├── [5774578]  5774571
+        |   └── [5774571]  example_file_in_containers.json
+        └── [5774573]  5774572
+            └── [5774572]  example_file_in_containers.json
+
+Usage:
+
+KEEP_TIME=3600 ./logs-keeper <directory_path>
+"""
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")
+log = logging.getLogger()
+KEEP_TIME = int(os.getenv("KEEP_TIME_S", 120))
```

Comment:
Is it `KEEP_TIME` or `KEEP_TIME_S`? Rel. line 78

Reply:
`KEEP_TIME_S`